### DISCUSSION TITLE
Suricata: rule mismatch + so-elastalert missing +

### DISCUSSION THREAD

**Discussion by ejgh-oe** [Security-Onion-Solutions/securityonion#13206] [2.4] [answered]
### Version

2.4.70

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

10

### RAM

32

### Storage for /

500G

### Storage for /nsm

500G

### Network Traffic Collection

span port

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

No, there are no additional clues

### Detail

Hi,
After upgrading from 2.4.60 -> 2.4.70 I had to reboot all nodes because OS-Updates were installed automatically waiting to be activating by rebooting all nodes (Status "Reboot" on the Grid display). After rebooting the manager mode didn't come online (Status "Fault").

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/762bb144-dce5-455e-839a-e1158d814d6f)

Rebooting the manager node didn't help - neither did `salt-call state.highstate`
```

# salt-call state.highstate
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/grains'
[INFO    ] Syncing grains for environment 'base'
[INFO    ] Loading cache from salt://_grains, for base
[INFO    ] Caching directory '_grains/' for environment 'base'
local:
    Data failed to compile:
----------
    The function "state.highstate" is running as PID 303805 and was started at 2                                           024, Jun 14 14:25:45.126449 with jid 20240614142545126449
#

```

Tried "salt-call state.highstate" several times over half an hour or so - same error message as above.

On the Detections tab I also got a "!" likely to be caused by this:

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/7b613fc9-85ea-4bbd-9ae4-d5713b9ffb97)

Clicking on "Suricata Rule Mismatch" brings me to the Hunt interface giving

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/e1262a3a-407c-4a9a-9286-dc0b472b9df6)

and scrolling to the right reveals

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/146d6c37-b573-4ea0-9969-f264eb392b66)

Interestingly enough the problems mentioned above didn't occur after upgrading from 2.4.60 > 2.4.70 - they started after rebooting all nodes as of today.

Thanks much in advance for any clue...

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by cm-ops** [comment]
If you click on the "Rule Mismatch", and filter for the "Integrity Check Report" log, what does it show?

Also, if so-elastalert is still showing missing, check your Elasticsearch cluster health - `sudo so-elasticsearch-query _cluster/health?pretty`

**Comment by ejgh-oe** [comment]
First of all the good news: the "so-elastalert"-problem seems to have disappeared over time - after several hours (basically overnight) so-elastalert was up&running again without any intervention from my side.

The "Rule Mismatch"-problem is still there though. Filtering for "Integrity Check Report" gives

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/d930bdc7-3067-4555-b7b4-66f4f67931a1)

and scrolling down further
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/a3611520-7886-4521-8319-54f8dd455e6f)

So these errors occur every couple of minutes

I exported one of the detail pages to json and found something very interesting:

There's a section `soc.fields.deployedButNotEnabled`

with the following content:
`
    "soc.fields.deployedButNotEnabled": [
        "9000008",
        "9000021",
        "9000020",
        "9000017",
        "9000016",
        "9000018",
        "9000002",
        "9000009",
        "9000013",
        "9000005",
        "9000004",
        "9000011",
        "9000012",
        "9000003",
        "9000007",
        "9000015",
        "9000019",
        "9000006",
        "9000010",
        "9000014"
    ],
`

These numbers seem to correspond to the SID-numbers of my local suricata rules.

These rules have been there for quite a while and I never had any problems with them. I configured then in 2.4.60 via Administration > Configuration > Advanced settings > idstools > rules > Local Rules and they've been working perfectly since then.

To me it seems like SecurityOnion is complaining about local rules, despite they seem to work OK (if the weren't working I had got tons of false positives).

Any clue as to what's going on here?

**Reply by defensivedepth** [reply, marked as answer]
Yes, this is a known bug that will be fixed in the upcoming release.

**Reply by ejgh-oe** [reply]
Perfect - thanks much.

**Reply by defensivedepth** [reply]
@ejgh-oe 2.4.80 was released yesterday and has a number of fixes for Detections. I would suggest upgrading and seeing if that fixes the issue.

**Reply by ejgh-oe** [reply]
Thanks for letting me know. I've already planned upgrading to 2.4.80 early next week - keep you posted as to the result.

**Reply by ejgh-oe** [reply]
Hi,
Unfortunately, upgrading to 2.4.80 didn't solve the problem - Suricata rule mismatch is still there flagging all my custom rules (range 9000000 and above); same behavior as the one I described in my postinv above.
Please let me know if you need any logs, screenshots etc. - anything that can help in tracking this down.

**Reply by defensivedepth** [reply]
Since you upgraded to .70, have you removed and added the rules back?

I think that is what you will need to do. You can do this individually through the Detection UI or add them to a new local file on the Manager and configure Detections to import them - more details about that option found here: https://docs.securityonion.net/en/2.4/nids.html#other

**Reply by ejgh-oe** [reply]
> Since you upgraded to .70, have you removed and added the rules back?

No - didn't change any rules after going from .70 to .80.

Originally (my installation started out with 2.4.40) I configured all custom Suricata-Rules via the GUI, i.e. Administration > Configuration > Advanced settings > idstools > rules > Local Rules. They ended up in `/opt/so/rules/nids/suri/local.rules` which where they're still.

> I think that is what you will need to do. You can do this individually through the Detection UI or add them to a new local file on the Manager and configure Detections to import them - more details about that option found here: https://docs.securityonion.net/en/2.4/nids.html#other

Given that all local suricata rules entered via the GUI are stored in `/opt/so/rules/nids/suri/local.rules` do you mean importing that file via something like

`{"community":true,"license":"DRL1.1","file":"/opt/so/rules/nids/suri/local.rules","ruleset":"SOS-Custom"}`

should do the trick?

Please note that my local rules refer to variables that I've set via `Administration > Configuration > (Show all configurable settings) > Suricata > Advanced`, so will these two, i.e. rules imported via the statement above and modifications for variables fit together?

**Reply by defensivedepth** [reply]
What's happening is that SOC Detections does not know about preexisting custom rules configured via  `/opt/so/rules/nids/suri/local.rules`. So you need to remove the rules in that file and then import them via the web interface in Detections or the local file option. Please doublecheck the doc link I posted - `file: This is the path for the local rules file, which must be in the /nsm/rules/detect-suricata/custom_file/ directory.`

Variables should not be a problem.

**Reply by defensivedepth** [reply]
To be clear, you actually need to remove your custom rules from the following location:

`/opt/so/saltstack/local/salt/idstools/rules/local.rules`

**Reply by ejgh-oe** [reply]
> To be clear, you actually need to remove your custom rules from the following location:

> `/opt/so/saltstack/local/salt/idstools/rules/local.rules`

Hi,

Easier said than done - I've been playing "whac-a-mole" with that file for quite a while now: removed the file and it magically reappeared after about 15 minutes without me touching anything config-wise in SO:

```
[root@onion-mgr suri]# date
Wed Jul  3 08:35:11 AM UTC 2024
[root@onion-mgr suri]# ls -lt
total 28900
-rw-r--r--. 1 socore socore 29558544 Jul  3 08:05 all.rules
-rw-r--r--. 1 socore socore    23554 Jul  3 08:04 local.rules
-rw-r--r--. 1 socore socore     2476 Jan 19 09:02 extraction.rules
-rw-r--r--. 1 socore socore     1102 Jan 19 09:02 filters.rules
[root@onion-mgr suri]# rm local.rules
rm: remove regular file 'local.rules'? y
[root@onion-mgr suri]# date
Wed Jul  3 08:35:21 AM UTC 2024
[root@onion-mgr suri]# pwd
/opt/so/rules/nids/suri
[root@onion-mgr suri]# 
```

...after 15 minutes it's here again:

```
[root@onion-mgr suri]# date
Wed Jul  3 08:51:03 AM UTC 2024
[root@onion-mgr suri]# ls -lt
total 28900
-rw-r--r--. 1 socore socore 29558544 Jul  3 08:50 all.rules
-rw-r--r--. 1 socore socore    23554 Jul  3 08:49 local.rules
-rw-r--r--. 1 socore socore     2476 Jan 19 09:02 extraction.rules
-rw-r--r--. 1 socore socore     1102 Jan 19 09:02 filters.rules
[root@onion-mgr suri]#
```

Since the content of this file came from the configuration I entered via the GUI under "pre 2.4.70" (i.e. _Administration > Configuration > Advanced settings > idstools > rules > Local Rules_ ) I tried to remove them via the GUI - unfortunately that isn't possible either:

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/84927703/98da52ce-6ca5-4a78-8403-fa2395c8ad55)

Any idea on how to get rid of the (old) local rules?

**Reply by ejgh-oe** [reply]
Finally figured it out: Deleting the old custom-rules file `/opt/so/rules/nids/suri/local.rules` doesn't help since it reappears every 15 minutes or so (see above).

The solution was to 

- edit the file
- delete any Suricata rules that are still in that file (so basically the file is either empty or you've got some comments left)
- save the file under it's original name

so that basically you've got an empty `/opt/so/rules/nids/suri/local.rules`

Once that is done you can add your custom rules either via the GUI method or the "include file"-method @defensivedepth described above.
Then give Suricata a full update as described at the bottom of https://docs.securityonion.net/en/2.4/nids.html#other and you should be good to go.

Maybe these tips will help someone running into the same problems I did.
